numpy/ma/extras.py

```diff
@@ -665,7 +620,19 @@ def median(a, axis=None, out=None, overwrite_input=False):
     """
     if not hasattr(a, 'mask') or np.count_nonzero(a.mask) == 0:
         return masked_array(np.median(getdata(a, subok=True), axis=axis,
-                      out=out, overwrite_input=overwrite_input), copy=False)
+                            out=out, overwrite_input=overwrite_input,
+                            keepdims=keepdims), copy=False)
+    if a is masked:
+        return masked
```

Comment:
This skips a bunch of error cases, so I am not the biggest fan, is it necessary?